src/ess/reduce/nexus/_nexus_loader.py

```diff
@@ -180,14 +222,25 @@ def load_event_data(
     Returns
     -------
     :
-        Data array with events grouped by event_time_zero, as in the NeXus file.
+        Data array with events or a histogram.
     """
     with _open_nexus_file(file_path, definitions=definitions) as f:
         entry = _unique_child_group(f, snx.NXentry, entry_name)
         instrument = _unique_child_group(entry, snx.NXinstrument, None)
         component = instrument[component_name]
-        event_data = _unique_child_group(component, snx.NXevent_data, None)
-        return event_data[selection]
+        if _contains_nx_class(component, snx.NXevent_data):
+            data = _unique_child_group(component, snx.NXevent_data, None)
+            sel = _to_snx_selection(selection, for_events=True)
+        elif _contains_nx_class(component, snx.NXdata):
+            data = _unique_child_group(component, snx.NXdata, None)
+            sel = _to_snx_selection(selection, for_events=False)
```

Comment:
If the monitor or detector is not time-dependent then there will not be an `NXdata`, just a HDF5 dataset.

Reply:
How do we identify the data in that case?

Reply:
ScippNexus does this, in can be a bit complicated in general, see https://github.com/scipp/scippnexus/blob/94779e6d5e10666eaccc7d343079f9762ef9e3ea/src/scippnexus/nxdata.py#L63-L69 as a starting point.

Reply:
This explains most of it: https://github.com/scipp/scippnexus/blob/94779e6d5e10666eaccc7d343079f9762ef9e3ea/src/scippnexus/nxdata.py#L116-L123

Reply:
So how about this?
```python
if _contains_nx_class(component, snx.NXevent_data):
    data = _unique_child_group(component, snx.NXevent_data, None)
    sel = _to_snx_selection(selection, for_events=True)
elif _contains_nx_class(component, snx.NXdata):
    data = _unique_child_group(component, snx.NXdata, None)
    sel = _to_snx_selection(selection, for_events=False)
elif _is_subclass_of_nxdata(component):
    data = component
    sel = _to_snx_selection(selection, for_events=False)
else:
    raise ValueError(
        f"NeXus group '{component.name}' contains neither "
        "NXevent_data nor NXdata."
    )
```
Or is `component` not required to be something compatible with NXdata?

Reply:
I think that might lead to loading the entire NXdetector nested, a second time?

Reply:
Depends on what `component` is. But yes, this can happen.

Can we just not support this case for now. The current implementation works for BIFROST. And we can always extend it later.